jerry-core/parser/js/js-parser-module.c

```diff
@@ -0,0 +1,803 @@
+      return true;
+    }
+    current = next;
+  }
+
+  return false;
+} /* parser_module_is_property_exported */
+
+/**
+ * Compare property name with imports.
+ * @return true - if the exported property is imported
+ *         false - otherwise
+ */
+static bool
+parser_module_compare_property_name_with_import (parser_module_node_t *module_node_p, /**< module node */
+                                                 parser_module_names_t *export_names_p, /**< export names */
+                                                 parser_module_names_t *eventual_names_p) /**< [out] used names */
+{
+  JERRY_UNUSED (eventual_names_p);
+
+  parser_module_names_t *current = module_node_p->module_names_p;
+  parser_module_names_t *next;
+
+  for (int i = 0; i < module_node_p->module_request_count; i++)
+  {
+    next = current->next_p;
+
+    if (current->local_name_length == export_names_p->import_name_length)
+    {
+      if (memcmp (export_names_p->import_name_p, current->local_name_p, current->local_name_length) == 0)
+      {
+        *eventual_names_p = *current;
+        return true;
+      }
+    }
+
+    current = next;
+  }
+
+  return false;
+} /* parser_module_compare_property_name_with_import */
+
+/**
+ * Run an EcmaScript module created by parser_module_parse.
+ */
+static void
+parser_module_run (parser_context_t *context_p, /**< parser context */
+                   const char *file_path_p, /**< file path */
+                   size_t path_size, /**< length of the path */
+                   const char *source_p, /**< module source */
+                   size_t source_size, /**< length of the source */
+                   parser_module_node_t *module_node_p) /**< module node */
+{
+  parser_module_node_t export_node;
+  jerry_value_t ret_value = parser_module_parse ((jerry_char_t *) file_path_p,
+                                                path_size,
+                                                (jerry_char_t *) source_p,
+                                                source_size,
+                                                JERRY_PARSE_STRICT_MODE,
+                                                module_node_p,
+                                                &export_node);
+  if (jerry_value_is_error (ret_value))
+  {
+    jerry_release_value (ret_value);
+    parser_raise_error (context_p, PARSER_ERR_MODULE_REQUEST_NOT_FOUND);
+  }
+
+  jerry_value_t func_val = ret_value;
+  ecma_object_t *func_obj_p = ecma_get_object_from_value (func_val);
+
+  if (ecma_get_object_type (func_obj_p) != ECMA_OBJECT_TYPE_FUNCTION)
+  {
+    parser_raise_error (context_p, PARSER_ERR_ARGUMENTS_NOT_ALLOWED);
+  }
+
+  ecma_object_t *scope_p = ecma_create_decl_lex_env (ecma_get_global_environment ());
+  ret_value = vm_run_module (ecma_op_function_get_compiled_code ((ecma_extended_object_t *) func_obj_p), scope_p);
+
+  ecma_object_t *global_lex_env_p = ecma_get_lex_env_binding_object (ecma_get_global_environment ());
```

Comment:
A lex_env's binding object cannot be a lex_env. The also the global lexenv's binding object is the builtin global object.

Reply:
Rename to `global_obj_p` and use `ecma_object_t *glob_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_GLOBAL);`